File: Лаб 4 ОП/Lab4CSharp/Lab4CSharp/Operations.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab4CSharp
{
    class Operations
    {
        public static double InputTime()
        {
            Console.Write("Enter t in seconds: ");
            while (true)
            {
                string input = Console.ReadLine();
                if (double.TryParse(input, out double time) && time > 0)
                {
                    return time;
                }
                else
                {
                    Console.Write("Wrong input. Try again: ");
                }
            }
        }

        public static int InputNumber(string message)
        {
            Console.Write($"Enter {message}: ({message} > 1) ");
            while (true)
            {
                string input = Console.ReadLine();
                if (int.TryParse(input, out int number) && number > 1)
                {
                    return number;
                }
                else
                {
                    Console.Write("Wrong input. Try again: ");
                }
            }
        }

        public static void ShowObjects(MovingPointParticle[] array)
        {
            for (int i = 0; i < array.Length; i++)
            {
                Console.Write($"{i + 1})");
                array[i].DisplayObject();
            }
        }

        public static void ShowCoordinates(MovingPointParticle[] array, double t)
        {
            for (int i = 0; i < array.Length; i++)
            {
                Console.Write($"{i + 1})");
                array[i].DisplayCoordinates(t);
            }
        }

        public static double ReturnMaxDistance(MovingPointParticle[] array, out int first, out int second, double t)
        {
            first = 0;
            second = 1;
            double maxDistance = array[0].GetDistance(array[1].GetCoordinates(t, out double y), y);
            for (int o = 0; o < array.Length - 1; o++)
            {
                for (int i = o + 1; i < array.Length; i++)
                {
                    double temp = array[o].GetDistance(array[i].GetCoordinates(t, out y), y);
                    if (temp > maxDistance)
                    {
                        maxDistance = temp;
                        first = o;
                        second = i;
                    }
                }
            }
            return maxDistance;
        }
    }
}

File: Лаб 4 ОП/Lab4Python/Operations.py
def input_time():
    time = float(input("Enter t in seconds: "))
    while True:
        if time > 0:
            return time
        else:
            time = float(input("Wrong input. Try again: "))


def input_number(message):
    number = int(input(f"Enter {message}: ({message} > 0) "))
    while True:
        if number > 0:
            return number
        else:
            number = float(input("Wrong input. Try again: "))


def show_objects(array):
    for i in range(len(array)):
        print(f"{i+1})", end="")
        array[i].display_object()


def show_coordinates(array, t):
    for i in range(len(array)):
        x, y = array[i].get_coordinates(t)
        print(f"{i+1})\tx is: \t{round(x, 3)}\ty is: \t{round(y, 3)}")


def return_max_distance(array):
    first = 0
    second = 1
    max_distance = array[0].get_distance(array[0].get_x(), array[0].get_y())
    for o in range(len(array)-1):
        for i in range(len(array)):
            temp_max = array[o].get_distance(array[i].get_x(), array[i].get_y())
            if temp_max > max_distance:
                max_distance = temp_max
                first = o
                second = i
    return max_distance, first, second



